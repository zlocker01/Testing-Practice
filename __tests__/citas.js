import Citas from "../js/classes/Citas";

describe('prboar la clase de citas', () => {
    const citas = new Citas();
    const id = Date.now();

    test('agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'panfilo',
            propietario: 'juan',
            telefono: '302',
            fecha: '34/96/1847',
            hora:'23:34',
            sintomas: 'dolor de panza'
        }

        citas.agregarCita(citaObj);
        //prueba
        expect(citas).toMatchSnapshot();
    });

    test ('actualizar cita', () => {
        const citaAtualziada = {
            id,
            mascota: 'dori el pez espada',
            propietario: 'pancrasio',
            telefono: '3029240942',
            fecha: '34/96/1847',
            hora:'23:34',
            sintomas: 'dolor de cornea xd'
        }

        citas.editarCita(citaAtualziada);

        expect(citas).toMatchSnapshot();
    })

    test('eliminar citas', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
})