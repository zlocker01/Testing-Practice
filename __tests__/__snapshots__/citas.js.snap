// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prboar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "34/96/1847",
      "hora": "23:34",
      "id": 1679092147130,
      "mascota": "dori el pez espada",
      "propietario": "pancrasio",
      "sintomas": "dolor de cornea xd",
      "telefono": "3029240942",
    },
  ],
}
`;

exports[`prboar la clase de citas agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "34/96/1847",
      "hora": "23:34",
      "id": 1679092147130,
      "mascota": "panfilo",
      "propietario": "juan",
      "sintomas": "dolor de panza",
      "telefono": "302",
    },
  ],
}
`;

exports[`prboar la clase de citas eliminar citas 1`] = `
Citas {
  "citas": [],
}
`;
